compiler/rustc_hir_typeck/src/callee.rs

```diff
@@ -297,6 +299,15 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                 return None;
             }
 
+            ty::Infer(ty::TyVar(vid)) => {
```

Comment:
feels subtle to me that we are specifically checking for `TyVar` instead of just all `ty::Infer`? How does the rest of the code handle int/float vars